### PR TITLE
Snippet choice support for enum and boolean.

### PR DESCRIPTION
**Pull request by angelozerr**
Snippet choice support for enum and boolean.

See #1

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
This PR fixes https://github.com/redhat-developer/vscode-quarkus/issues/25 and provide enum completion (only when key completion is applied).

QuakusCompletionChoixe

**Comment by xorye** [comment]
Great PR, I very much like the text edits to deal with https://github.com/redhat-developer/vscode-quarkus/issues/25

I have some comments:


The snippet choice works great, but when I select one of the options (in this case, `BLOCK`), and then If I hit "Enter" to go to the next line, completion does not show up as I type. There is a grey highlight covering the text that I am typing. I think this highlight is preventing me from getting completion options.\
\
\

The grey highlight goes away when I hit "Tab" however. Once the grey highlight is gone, completion works as I type.\
\
\

However, despite having grey highlights, the completion options still show up if you invoke completion with "Ctrl + Space". The problem is that the grey highlights prevent completion options showing as you type.\
\
I'm not sure if this is something that can be fixed. Is this something we should be concerned about?

File: quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusCompletions.java [hunk 1]
```diff
@@ -88,26 +98,62 @@ private static void collectPropertyKeySuggestions(TextDocument document, Propert
...
+			insertText.append(' '); // TODO: this space should be configured in format settings
+			insertText.append('=');
+			insertText.append(' '); // TODO: this space should be configured in format settings
+
+			if (enums != null) {
+				// Enumerations
+				if (snippetsSupported) {
+					SnippetsBuilder.choice(1, enums, insertText);
+				} else {
+
```
Comment (xorye) [inline quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusCompletions.java]: Just double checking, was there supposed to be something here?

**Comment by angelozerr** [comment]
Thanks @xorye for your feedback, I will study more your issue.

**Comment by fbricon** [comment]
When you hit enter, you're still in "snippet" mode in vscode, as you're inside a placeholder, in that mode, completion is not called. If you hit `Esc` after selecting your value, it'll work. Long story short, this is a VS Code problem, orthogonal to this PR.

**Comment by angelozerr** [comment]
@fbricon can we merge this PR?

File: quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusCompletions.java (hunk 1 above)
Comment (angelozerr) [reply quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusCompletions.java]: Good catch. I fix it and add tests for snippetSupport or not.

File: quarkus.ls/com.redhat.quarkus.ls/src/test/java/com/redhat/quarkus/services/ApplicationPropertiesHoverTest.java [hunk 2]
```diff
@@ -26,35 +26,35 @@
  */
 public class ApplicationPropertiesHoverTest {
 
-	@Test
```
Comment (fbricon) [inline quarkus.ls/com.redhat.quarkus.ls/src/test/java/com/redhat/quarkus/services/ApplicationPropertiesHoverTest.java]: why?

Comment (angelozerr) [reply quarkus.ls/com.redhat.quarkus.ls/src/test/java/com/redhat/quarkus/services/ApplicationPropertiesHoverTest.java]: Sorry with my mistake, it's fixed.

**Pull request merged**